plugins/lookup/kustomize.py

```diff
@@ -0,0 +1,144 @@
+      opt_dirs:
+        description:
+        - optional list of directories to search for executable in addition to PATH.
+
+    requirements:
+      - "python >= 3.6"
+'''
+
+EXAMPLES = """
+- name: Run lookup using kustomize
+  set_fact:
+    resources: "{{ lookup('kustomize', binary_path='/path/to/kustomize') }}"
+
+- name: Run lookup using kubectl kustomize
+  set_fact:
+    resources: "{{ lookup('kustomize', binary_path='/path/to/kubectl') }}"
+
+- name: Create kubernetes resources for lookup output
+  k8s:
+    definition: "{{ item }}"
+  with_items:
+    "{{ lookup('kustomize', dir='/path/to/kustomization') }}"
+"""
+
+RETURN = """
+  _list:
+    description:
+      - One ore more object definitions returned from the tool execution.
+    type: complex
+    contains:
+      api_version:
+        description: The versioned schema of this representation of an object.
+        returned: success
+        type: str
+      kind:
+        description: Represents the REST resource this object represents.
+        returned: success
+        type: str
+      metadata:
+        description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+        returned: success
+        type: complex
+      spec:
+        description: Specific attributes of the object. Will vary based on the I(api_version) and I(kind).
+        returned: success
+        type: complex
+      status:
+        description: Current status details for the object.
+        returned: success
+        type: complex
+"""
+
+from ansible.errors import AnsibleLookupError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils.common.process import get_bin_path
+from ansible.module_utils._text import to_native
+
+
+import os
+import subprocess
+
+
+def get_binary_from_path(name, opt_dirs=None):
+    try:
+        opt_arg = {}
+        if opt_dirs is not None:
+            if not isinstance(opt_dirs, list):
+                opt_dirs = [opt_dirs]
+            opt_arg['opt_dirs'] = opt_dirs
+        bin_path = get_bin_path(name, **opt_arg)
+        return bin_path
+    except ValueError:
+        return None
+
+
+def run_command(command):
+    cmd = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    return cmd.communicate()
+
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, variables=None, dir=".", binary_path=None, opt_dirs=None, **kwargs):
+        try:
+            executable_path = binary_path
+            if executable_path is None:
+                executable_path = get_binary_from_path(name="kustomize", opt_dirs=opt_dirs)
+                if executable_path is None:
+                    executable_path = get_binary_from_path(name="kubectl", opt_dirs=opt_dirs)
+
+                # validate that at least one tool was found
+                if executable_path is None:
+                    raise AnsibleLookupError("Failed to find required executable 'kubectl' and 'kustomize' in paths")
```

Comment:
I think we can move this out of the try-except block.

Reply:
yes agree. Thanks